olive/passes/onnx/transformer_optimization.py

```diff
@@ -93,7 +94,7 @@ def _run_for_config(self, model: ONNXModel, config: Dict[str, Any], output_model
         for key in get_external_data_config():
             del run_config[key]
 
-        output_model_path = ONNXModel.resolve_path(output_model_path)
+        output_model_path = ONNXModel.resolve_path(os.path.join(output_model_path, os.path.basename(model.model_path)))
```

Comment:
Is there a reason for this change? Ideally, we don't want to manipulate the output path much other than adding a suffix or creating a file in the directory. Since it might affect the caching of the model.

Reply:
The default resulting model name of `resolve_path` is simply `model.onnx`, no matter what the original model is named.

It works for pipelines with a single model, but for pipelines with multiple models (e.g. `decoder_model.onnx` and `decoder_with_past_model.onnx`), it makes it very hard to debug intermediate steps, or even use the result of intermediate steps for something else. All this change does is name the optimized model the same way that the unoptimized version is named.

Reply:
I see. Thanks for the clarification. For the other pass where we have composite models, we create a subdirectory for each component so they are in different folders. 
I think this change is okay for now since the file is still in the parent directory of the `output_model_path`. I will revisit this later to standardize how the paths are manipulted.

Reply:
I agree with @PatriceVignola. The output model path might be modified based on the arguments. For example, the output model path might be appended with hardware info if the pass decide it is hardware dependent like https://github.com/microsoft/Olive/pull/286/files#diff-89a2a3c72dc004cdf3c6603d382f434ec3f7527fc9065a7f3c420ceaf3a7cbf9R110-R111